### PR TITLE
Added docs for extraObjcCompilerArgs

### PR DESCRIPTION
**Pull request by confile**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,10 @@ Tracked issue: https://github.com/j2objc-contrib/j2objc-gradle/issues/130**
     j2objcConfig {
         xcodeProjectDir "${projectDir}/../ios"   // Xcode workspace directory
         xcodeTarget "MyTarget"                   // Xcode application target name
+        
+        // make ARC avalable for the objective-c compiler
+        extraObjcCompilerArgs += '-fobjc-arc'
```
Comment (brunobowden) [inline README.md]: Having this in the default example is basically suggesting making ARC the default - do we want to do this?.

I'm wondering if we should comment this out, so people don't enable unless they clearly know that they need it.

**Comment by brunobowden** [comment]
@advayDev1 - interested in your thoughts on this

**Comment by advayDev1** [comment]
my view is consistency.  since the default J2objc does not have -use-arc, i don't see why the default compile would have -fobjc-arc.

**Comment by confile** [comment]
Done

**Comment by confile** [comment]
Swift uses ARC per default so we should consider doing this as well.

**Comment by brunobowden** [comment]
The J2ObjC team recommeend NOT using ARC. If you do a search for
"j2objc-discuss" and ARC, you'll find the following from late last year:

Keith Stanger:
"My opinion on this may be different than Tom's, but I always advise users
not to use ARC with translated code."

https://groups.google.com/forum/#!topic/j2objc-discuss/sGJHQgao788

**Comment by confile** [comment]
Okay lets keep it. It is easy to set the flag. Lets focus on the other issues.

File: README.md [hunk 2]
```diff
@@ -36,6 +36,10 @@ Tracked issue: https://github.com/j2objc-contrib/j2objc-gradle/issues/130**
     j2objcConfig {
         xcodeProjectDir "${projectDir}/../ios"   // Xcode workspace directory
         xcodeTarget "MyTarget"                   // Xcode application target name
+        
+        // make ARC avalable for the objective-c compiler
+        // extraObjcCompilerArgs += '-fobjc-arc'
```
Comment (brunobowden) [inline README.md]: I'd remove this from the README.md and have it solely in the FAQ.

**Comment by brunobowden** [comment]
@tomball, @kstanger - looking at the J2ObjC docs, it appears that you now recommend using ARC for new projects (this is different to what Keith suggested on j2objc-discuss in the past). Accordingly, we're thinking of suggesting ARC for the j2objc-gradle plugin.

Will this become a default for the j2objc? This is important to us as the plugin has the convention of defaulting to the j2objc command line flag defaults.

Current documentation:
http://j2objc.org/docs/Memory-Management.html#automatic-reference-counting-arc

@confile - let's hold off until we get Tom or Keith's insights.

**Comment by confile** [comment]
@brunobowden Agreed!

**Comment by kstanger** [comment]
Note that j2objc's -use-arc and clang's -fobjc-arc go hand-in-hand. If you use one without the other compilation will fail.

My opinion remains that translated code is best without ARC.

**Comment by brunobowden** [comment]
Understood Keth. My concern is that should have consistency between the
documentation, j2objc command line, j2objc-discuss and the gralde plugin.
It's quite alright to present alternatives but we should collectively agree
on the default.

**Comment by advayDev1** [comment]
The issue @kstanger mentioned is already fired as a warning by implementation #185 but perhaps it should become an error.

File: README.md [hunk 3]
```diff
@@ -36,6 +36,8 @@ Tracked issue: https://github.com/j2objc-contrib/j2objc-gradle/issues/130**
     j2objcConfig {
         xcodeProjectDir "${projectDir}/../ios"   // Xcode workspace directory
         xcodeTarget "MyTarget"                   // Xcode application target name
```
Comment (brunobowden) [inline README.md]: Match other documentation:

    xcodeTarget "IosApp"                   // Xcode application target name

File: README.md [hunk 4]
```diff
@@ -36,6 +36,8 @@ Tracked issue: https://github.com/j2objc-contrib/j2objc-gradle/issues/130**
...
+
```
Comment (brunobowden) [inline README.md]: Only one blank line rather than two

File: README.md [hunk 5]
```diff
@@ -36,6 +36,8 @@ Tracked issue: https://github.com/j2objc-contrib/j2objc-gradle/issues/130**
     j2objcConfig {
         xcodeProjectDir "${projectDir}/../ios"   // Xcode workspace directory
         xcodeTarget "MyTarget"                   // Xcode application target name
+    
+
         // More: https://github.com/j2objc-contrib/j2objc-gradle/blob/master/src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy#L25
```
Comment (brunobowden) [inline README.md]: Amend doc link to:

```
        // Documentation on further configuration:
        // https://github.com/j2objc-contrib/j2objc-gradle/blob/master/src/main/groovy/com/github/j2objccontrib/j2objcgradle/J2objcPluginExtension.groovy#L25
```

File: FAQ.md [hunk 6]
```diff
@@ -18,3 +18,18 @@ sourceSets {
...
+### How do I enable ARC for my objective-c classes?
+
+Add the following to your configuration block.
+
+```
+j2objcConfig {
+   extraObjcCompilerArgs += '-fobjc-arc'
+}
+```
+
```
Comment (advayDev1) [inline FAQ.md]: Add a reference to https://developer.apple.com/library/mac/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226-CH1-SW15

(and keep the blank line too)

**Comment by advayDev1** [comment]
LGTM from me, after doing the above commits (+squash/rebase)

File: FAQ.md (hunk 6 above)
Comment (confile) [inline FAQ.md]: done

File: README.md (hunk 3 above)
Comment (confile) [inline README.md]: What is your point here?

File: README.md (hunk 4 above)
Comment (confile) [inline README.md]: done

File: README.md (hunk 5 above)
Comment (confile) [inline README.md]: don't get it?

**Comment by confile** [comment]
@advayDev1 done

File: FAQ.md [hunk 7]
```diff
@@ -22,3 +22,18 @@ sourceSets {
...
 ```
+
+
+### How do I enable ARC for my objective-c classes?
+
+Add the following to your configuration block. [See](https://developer.apple.com/library/mac/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html#//apple_ref/doc/uid/TP40011226-CH1-SW15).
+
+```
+j2objcConfig {
+   extraObjcCompilerArgs += '-fobjc-arc'
```
Comment (brunobowden) [inline FAQ.md]: Include the translateFlags line as well

File: README.md (hunk 3 above)
Comment (brunobowden) [inline README.md]: To match the documentation in README.md

Comment (confile) [inline README.md]: Still don'T get what I should change!??!

Comment (brunobowden) [inline README.md]: This is a comment line 39. Change that to new code that I gave. Note how "MyTarget" becomes "IosApp" in the comment, repeated here:

    xcodeTarget "IosApp"                   // Xcode application target name

Comment (confile) [inline README.md]: done

**Comment by confile** [comment]
done

File: FAQ.md [hunk 8]
```diff
@@ -22,3 +22,19 @@ sourceSets {
...
+   translateFlags '-use-arc'
```
Comment (brunobowden) [inline FAQ.md]: Swap two lines to match execution order... putting the translate before the compile args. This also matches the code error.

File: README.md [hunk 9]
```diff
@@ -36,7 +36,8 @@ Tracked issue: https://github.com/j2objc-contrib/j2objc-gradle/issues/130**
     // Plugin settings:
     j2objcConfig {
         xcodeProjectDir "${projectDir}/../ios"   // Xcode workspace directory
-        xcodeTarget "MyTarget"                   // Xcode application target name
+        xcodeTarget "IosApp"                   // Xcode application target name
```
Comment (brunobowden) [inline README.md]: NIT: indent the comment two spaces so it matches the line above.

File: FAQ.md (hunk 8 above)
Comment (confile) [inline FAQ.md]: done

File: FAQ.md (hunk 7 above)
Comment (confile) [inline FAQ.md]: done

**Comment by confile** [comment]
done

**Pull request merged**